tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -602,9 +604,17 @@ HRESULT STDMETHODCALLTYPE CorProfiler::ModuleLoadFinished(ModuleID module_id, HR
                                                                                 &promise);
 
             // wait and get the value from the future<ULONG>
-            const auto& numReJITs = future.get();
+            const auto status = future.wait_for(100ms);
 
-            Logger::Debug("Total number of ReJIT Requested: ", numReJITs);
+            if (status == std::future_status::ready)
+            {
+                const auto& numReJITs = future.get();
+                Logger::Debug("Total number of ReJIT Requested: ", numReJITs);    
+            }
+            else
+            {
+                Logger::Warn("Timeout while waiting for the rejit requests to be processed");
```

Comment:
Is it worth printing the implications here, to avoid/encourage scaring people? e.g. something like:

```suggestion
                Logger::Warn("Timeout while waiting for the rejit requests to be processed. Rejit will continue asynchronously, but some initial calls may not be instrumented");
```